lib/semantics/check-do-stmt.cc

```diff
@@ -344,41 +372,108 @@ static CS GatherReferencesFromExpression(const parser::Expr &expression) {
   }
 }
 
-// Find a canonical DO CONCURRENT and enforce semantics checks on its body
-class DoConcurrentContext {
+// Find a DO statement and enforce semantics checks on its body
+class DoStmtContext {
 public:
-  DoConcurrentContext(SemanticsContext &context)
-    : messages_{context.messages()} {}
+  DoStmtContext(SemanticsContext &context)
+    : context_{context}, messages_{context.messages()} {}
 
-  bool operator==(const DoConcurrentContext &x) const { return this == &x; }
+  bool operator==(const DoStmtContext &x) const { return this == &x; }
```

Comment:
I think the `operator==` was only needed because of the use of this class in `Indirection<>` so this is probably not necessary anymore.

Reply:
Apparently so.  I was able to delete it with no ill effects.